### ISSUE TITLE
Wix Extension Custom Actions not signed

### ISSUE THREAD

**Issue by chrpai** [wixtoolset/issues#7163]
4.0 rc2 build 23
vs 2022 17.4.3
Heatwave 0.9.3

I had observed in a rc2 build14 that the custom actions were versioned as 4.0.0.0 and that WiX Toolset 4.0.0.0 was in the MSI Summary Information Stream.  The DLLs in the Binary table were signed though.  I then updated to rc2 build 23 to compare and I found that the version numbers were all good but the DLLs were not signed.

I'm not sure how this gets built and published or why one build was signed and another wasn't so I just wanted to point this out.

**Comment by robmen** [comment]
Dev builds (builds that end with `-build.#`) are not signed. I spot-checked a couple of .wixexts (UI, Util) from `-rc.2-build.14` and they were not signed.

Let me know which exact files from a dev build are signed or have incorrect versions. At this point, things seem to be correct.

**Comment by chrpai** [comment]
![image](https://user-images.githubusercontent.com/43178814/213054402-0f26d3a8-bdc6-4ca3-a301-6e72eb1aeed2.png)
![image](https://user-images.githubusercontent.com/43178814/213054424-4c780d92-d6f4-4089-8ac7-2dff56518694.png)

**Comment by robmen** [comment]
Interesting. What dev build is that from? I downloaded Iis.wixext `-rc.2-build.14`, and it looks okay. I'm a little confused about which build and .wixexts I should look at to find the invalid version numbers and signed files.

**Comment by chrpai** [comment]
That's extracted from the binary table of an MSI built with 14.  See it says WiXToolset IIS Custom Action and the original filename is iisca.dll.

**Comment by robmen** [comment]
Okay, I figured it out. The images above are of the IIS CA from the official 4.0.0-rc.1 build. The "Product version" is `4.0.0-rc.1+6580fe...` and `+6580fe...` is the git commit id from the official -rc.1 build. That explains the version numbers and why the binaries are signed.

**Comment by chrpai** [comment]
Hmmm, I'm pretty certain that I had the wixproj  versions set to 14 when I built that MSI as I had it set to 23 when I built it a second time.  Maybe I didn't?  As long as official release are versioned and signed then it's alright.  In an agile nuget world these version numbers become pretty important.

![image](https://user-images.githubusercontent.com/43178814/213350005-6e243812-76ec-4014-8739-7c47c26cb24b.png)
